From a9fcbdfaf4c553e5b5bfbd866734798581f76aea Mon Sep 17 00:00:00 2001
From: dponna534 <Deepak_Ponnath2@comcast.com>
Date: Mon, 1 Sep 2025 16:30:50 +0530
Subject: [PATCH] RDKEMW-3659: Get model implemenatation change.

Reson for change: Use device info plugin to read model id
Test Procedure: Manual
Risks: Low

Signed-off-by: Deepak_Ponnath2@comcast.com
---
 SystemServices/SystemServices.cpp | 34 ++++++++++++++++++++++++++++++-
 1 file changed, 33 insertions(+), 1 deletion(-)

diff --git a/SystemServices/SystemServices.cpp b/SystemServices/SystemServices.cpp
index a4cc483..b13ef1a 100755
--- a/SystemServices/SystemServices.cpp
+++ b/SystemServices/SystemServices.cpp
@@ -70,6 +70,12 @@
 #include "UtilsgetFileContent.h"
 #include "UtilsProcess.h"
 
+#ifdef USE_THUNDER_R4
+#include <interfaces/IDeviceInfo.h>
+#else
+#include <interfaces/IDeviceInfo2.h>
+#endif /* USE_THUNDER_R4 */
+
 using namespace std;
 using namespace WPEFramework;
 using PowerState = WPEFramework::Exchange::IPowerManager::PowerState;
@@ -2456,7 +2462,7 @@ namespace WPEFramework {
             std::string estbMac = collectDeviceInfo("estb_mac");
             removeCharsFromString(estbMac, "\n\r");
             rConf["eStbMac"] = estbMac;
-            rConf["model"] = getModel();
+            rConf["model"] = RetrieveModelNumberThroughCOMRPC();
             rConf["firmwareVersion"] = stbVersion;
             response["xconfParams"] = rConf;
             returnResponse(true);
@@ -4443,6 +4449,32 @@ namespace WPEFramework {
             return "unknown";
 #endif
         }
+
+        string SystemServices::RetrieveModelNumberThroughCOMRPC()
+        {
+            LOGINFO("Systemservices: RetrieveModelNumberThroughCOMRPC Entry\n");
+            std::string Number;
+            if (m_shellService)
+            {
+                PluginHost::IShell::state state;
+
+
+                    auto _remoteDeviceInfoObject = m_shellService->QueryInterfaceByCallsign<Exchange::IDeviceInfo>("DeviceInfo");
+
+                    if (_remoteDeviceInfoObject)
+                    {
+                        _remoteDeviceInfoObject->Sku(Number);
+                        _remoteDeviceInfoObject->Release();
+                    }
+                    else
+                    {
+                        LOGERR("Failed to create DeviceInfo object\n");
+                    }
+                
+            }
+            return Number;
+        }
+
 	string SystemServices::getStbBranchString()
 	{
 		static string stbBranchStr;
-- 
2.47.1.windows.2

